google/cloud/spanner_v1/session.py

```diff
@@ -458,47 +463,100 @@ def run_in_transaction(self, func, *args, **kw):
         )
         attempts = 0
 
-        while True:
-            if self._transaction is None:
-                txn = self.transaction()
-                txn.transaction_tag = transaction_tag
-                txn.exclude_txn_from_change_streams = exclude_txn_from_change_streams
-            else:
-                txn = self._transaction
+        observability_options = getattr(self._database, "observability_options", None)
+        with trace_call(
+            "CloudSpanner.Session.run_in_transaction",
+            self,
+            observability_options=observability_options,
+        ) as span:
+            while True:
+                if self._transaction is None:
+                    add_span_event(span, "Creating Transaction")
```

Comment:
This is an inmemory transaction and not actual transaction call to backend. This might create confusion.
Lets remove this event.

Reply:
Gotcha and thanks!

Reply:
Done.